dask/dataframe/rolling.py

```diff
@@ -34,50 +33,106 @@ def overlap_chunk(
 
     parts = [p for p in (prev_part, current_part, next_part) if p is not None]
     combined = methods.concat(parts)
-    out = func(combined, *args, **kwargs)
-    if prev_part is None:
+
+    return CombinedOutput(
+        (
+            combined,
+            len(prev_part) if prev_part is not None else None,
+            len(next_part) if next_part is not None else None,
+        )
+    )
+
+
+def overlap_chunk(func, before, after, *args, **kwargs):
+    dfs = [df for df in args if isinstance(df, CombinedOutput)]
+    combined, prev_part_length, next_part_length = dfs[0]
+
+    args = [arg[0] if isinstance(arg, CombinedOutput) else arg for arg in args]
+
+    out = func(*args, **kwargs)
+
+    if prev_part_length is None:
         before = None
     if isinstance(before, datetime.timedelta):
-        before = len(prev_part)
+        before = prev_part_length
 
     expansion = None
     if combined.shape[0] != 0:
         expansion = out.shape[0] // combined.shape[0]
     if before and expansion:
         before *= expansion
-    if next_part is None:
+    if next_part_length is None:
         return out.iloc[before:]
     if isinstance(after, datetime.timedelta):
-        after = len(next_part)
+        after = next_part_length
     if after and expansion:
         after *= expansion
     return out.iloc[before:-after]
 
 
-def map_overlap(func, df, before, after, *args, **kwargs):
+def map_overlap(
+    func,
+    before,
+    after,
+    *args,
+    meta=no_default,
+    enforce_metadata=True,
+    transform_divisions=True,
+    align_dataframes=True,
+    **kwargs,
+):
```

Comment:
Re https://github.com/dask/dask/issues/9042#issuecomment-1122353218, 

@ian-r-rose and I think we should prioritize API stability over consistency with `map_partitions` here. Even though this is a good change, let's not break the old code?

Reply:
> I don't know how API changes are managed in Dask.

Dask doesn't have strong policies here, but usually, we have a Deprecation cycle. Since this is a big change, maybe a 2-month long cycle. For example: #9140 and #8907

For this PR, here's what this might look like:

```python
def map_overlap(
    func,
    before,
    after,
    *args,
    meta=no_default,
    enforce_metadata=True,
    transform_divisions=True,
    align_dataframes=True,
    **kwargs,
):
    if isinstance(before, _Frame):
        before = after
        after = args[0]
        args2 = [before, *args[1:]]
        warnings.warn ("The Signature will soon change to ..", FutureWarning)
        return _map_overlap(func, before, after, *args2, meta=meta, enforce_metadata=enforce_metadata, transform_divisions=transform_divisions, **kwargs)
    else:
        return _map_overlap(func, before, after, *args, meta=meta, enforce_metadata=enforce_metadata, transform_divisions=transform_divisions, **kwargs)


def _map_overlap(
    func,
    before,
    after,
    *args,
    meta=no_default,
    enforce_metadata=True,
    transform_divisions=True,
    align_dataframes=True,
    **kwargs,
):
    # everything currently in `map_overlap()` in this PR
    pass

```

This still would cover a case like referring to `before` and `after` as keyword arguments: `map_overlap(func, df, after=1, before=3)`. @ian-r-rose and I are not sure if this is worth it, that's why we suggested not changing it.

With this context, do you still think we should change the signature?

Reply:
Ok to keep the existing API, it was more a "API concistency" request,
I rollback the API changes in my last commit.